hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-distributedshell/src/main/java/org/apache/hadoop/yarn/applications/distributedshell/Client.java

```diff
@@ -1179,7 +1180,7 @@ private boolean monitorApplication(ApplicationId appId)
 
       // The value equal or less than 0 means no timeout
       if (clientTimeout > 0
-          && System.currentTimeMillis() > (clientStartTime + clientTimeout)) {
+          && Time.monotonicNow() > (clientStartTime + clientTimeout)) {
```

Comment:
This can break if clientStartTime + clientTimeout rolls over to a negative value.